workshops/models.py

```diff
@@ -397,6 +400,42 @@ def is_staff(self):
     def is_admin(self):
         return is_admin(self)
 
+    @property
+    def missing_swc_instructor_requirements(self):
+        """ Returns set of requirements that are not passed yet by the
+        trainee and are mandatory to become SWC Instructor.. """
+        training = TrainingRequirement.objects.get(name='Training')
+        homework = TrainingRequirement.objects.get(name='SWC Homework')
+        discussion = TrainingRequirement.objects.get(name='Discussion')
+        demo = TrainingRequirement.objects.get(name='SWC Demo')
+
+        requirements = {training, homework, discussion, demo}
+        progresses = self.trainingprogress_set.filter(discarded=False, state='p')
+        passed_requirements = {p.requirement for p in progresses}
+        return requirements - passed_requirements
```

Comment:
Then you should not make a property out of this helper function.

I talked to you and Greg about the root issue here and the response was "don't try to pre-optimize". You see, if when person is added to the training event, they should be assigned "empty" requirements upfront. Then you'd have no troubles checking which requirements were fulfilled and which were not. (So this is like copying empty requirements when people are being allowed/invited to the training.)